### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by miguelgaio**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/245%23issuecomment-142218412%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23issuecomment-142291132%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40079670%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40079753%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40085979%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40085981%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/245%23issuecomment-143162023%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/245%23issuecomment-142218412%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22The%20first%20three%20commits%20are%20already%20present%20on%20the%20windows_port%20branch.%22%2C%20%22created_at%22%3A%20%222015-09-22T08%3A57%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-22T13%3A35%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22Close.%20KW%20will%20be%20treated%20in%20other%20PR%22%2C%20%22created_at%22%3A%20%222015-09-25T08%3A41%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2097635f793697099b236b05a4b90188d5c513958f%20parameter/ParameterMgrLogger.h%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40079753%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%20-%20%60%3D%20default%60%20%3F%5Cr%5Cn%20-%20Final%20instead%20%3F%5Cr%5Cn%20-%20I%20think%20that%20the%20base%20class%20should%20be%20virtual%20also%20if%20this%20is%20virtual.%22%2C%20%22created_at%22%3A%20%222015-09-22T12%3A30%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-22T13%3A35%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgrLogger.h%3AL45-52%22%7D%2C%20%22Pull%2097635f793697099b236b05a4b90188d5c513958f%20test/test-platform/TestPlatform.cpp%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/245%23discussion_r40079670%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20this%20still%20needed%20as%20the%20class%20is%20final%20%3F%22%2C%20%22created_at%22%3A%20%222015-09-22T12%3A29%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-22T13%3A35%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-platform/TestPlatform.cpp%3AL41-52%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/245#issuecomment-142218412'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> The first three commits are already present on the windows_port branch.
- <a href='https://github.com/miguelgaio'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/miguelgaio'></a> Close. KW will be treated in other PR
- [x] <a href='#crh-comment-Pull 97635f793697099b236b05a4b90188d5c513958f test/test-platform/TestPlatform.cpp 9'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/245#discussion_r40079670'>File: test/test-platform/TestPlatform.cpp:L41-52</a></b>
- <a href='https://github.com/krocard'></a> Is this still needed as the class is final ?
- [x] <a href='#crh-comment-Pull 97635f793697099b236b05a4b90188d5c513958f parameter/ParameterMgrLogger.h 4'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/245#discussion_r40079753'>File: parameter/ParameterMgrLogger.h:L45-52</a></b>
- <a href='https://github.com/krocard'></a> - `= default` ?
- Final instead ?
- I think that the base class should be virtual also if this is virtual.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/245?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/245?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/245'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/245'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
The first three commits are already present on the windows_port branch.

File: test/test-platform/TestPlatform.cpp [hunk 1]
```diff
@@ -41,10 +41,11 @@
 
 using std::string;
 
-class CParameterMgrPlatformConnectorLogger : public CParameterMgrPlatformConnector::ILogger
+class CParameterMgrPlatformConnectorLogger final : public CParameterMgrPlatformConnector::ILogger
 {
 public:
     CParameterMgrPlatformConnectorLogger() {}
+    virtual ~CParameterMgrPlatformConnectorLogger() {}
```
Comment (krocard) [inline test/test-platform/TestPlatform.cpp]: Is this still needed as the class is final ?

File: parameter/ParameterMgrLogger.h [hunk 2]
```diff
@@ -45,6 +45,7 @@ class CParameterMgrLogger : public core::log::ILogger, private utility::NonCopya
         _parameterMgrConnector(parameterMgrConnector)
     {
     }
+    virtual ~CParameterMgrLogger() {}
```
Comment (krocard) [inline parameter/ParameterMgrLogger.h]: Final instead ?

Comment (miguelgaio) [inline parameter/ParameterMgrLogger.h]: :+1:<a href='#crh-update-none'></a>

**Comment by miguelgaio** [comment]
:+1:<a href='#crh-update-none'></a>

File: test/test-platform/TestPlatform.cpp (hunk 1 above)
Comment (miguelgaio) [inline test/test-platform/TestPlatform.cpp]: :+1:<a href='#crh-update-complete'></a>

**Comment by miguelgaio** [comment]
Close. KW will be treated in other PR

**Pull request closed without merging**